CanvasColorSpaceProposal.md

```diff
@@ -164,40 +159,55 @@ Note: An alternative approach that was considered was to augment the probablySup
 
 #### ImageBitmap
 
-TODO(ccameron): Review this
-ImageBitmap objects are augmented to have an internal color space attribute of type CanvasColorSpace and an internal pixelFormat attribute of type CanvasPixelFormat. The colorSpaceConversion creation attribute also accepts enum values that correspond to CanvasColorSpace values. Specifying a CanvasColorSpace value results in a conversion of the image to the specified color space.
+ImageBitmap objects (unless created with ``colorSpaceConversion="none"``) should keep track of their internal color space, and should store their contents at highest fidelity possible, subject to implementation limitations.
 
 #### ImageData
 
-TODO(ccameron): Review this
-
-IDL
+Add the following types to be used by `ImageData`.
 <pre>
-
-enum ImageDataStorageType {
-  "uint8", // default
-  "uint16",
+enum ImageDataEncoding {
+  "unorm8", // default
+  "unorm16",
   "float32",
 };
-
+dictionary ImageDataSettings {
+  CanvasColorSpaceEnum colorSpace = "srgb";
+  ImageDataStorageType encoding = "unorm8";
+};
 typedef (Uint8ClampedArray or Uint16Array or Float32Array) ImageDataArray;
+</pre>
 
-[Constructor(unsigned long sw, unsigned long sh, optional ImageDataColorSettings imageDataColorSettings),
- Constructor(ImageDataArray data, unsigned long sw, optional unsigned long sh, optional ImageDataColorSettings imageDataColorSettings),
- Exposed=(Window,Worker)]
-interface ImageData {
-  readonly attribute unsigned long width;
-  readonly attribute unsigned long height;
+Update the `ImageData` interface to the include the following.
+<pre>
+partial interface ImageData {
+  constructor(unsigned long sw, unsigned long sh, optional ImageDataSettings);
+  constructor(ImageDataArray data, unsigned long sw, unsigned long sh, optional ImageDataSettings);
+  readonly ImageDataSettings getImageDataSettings();
   readonly attribute ImageDataArray data;
-
-  ImageDataColorSettings getColorSettings();
 };
 </pre>
 
-* When using the constructor that takes an ImageDataArray parameter, the "storageType" setting is ignored.
-* createImageData() and getImageData() produce an ImageData object with the same color space as the source canvas, using an ImageDataArray of a type that is appropriate for the pixelFormat of the source canvas (smallest possible numeric size that guarantees no loss of precision).
-* putImageData() performs a color space conversion to the color space of the destination canvas.
-* Data returned by getImageData() or passed to putImageData() are assumed to be in "srgb" color space, with "unorm8" encoding.
+The changes to this interface are:
+* The constructors now take an optional `ImageDataSettings` dictionary.
+* The ImageDataSettings attribute may be queried using `getImageDataSettings`.
+* The constructor and attribute that used to be a `Uint8ClampedArray` are now a `ImageDataArray` union, which can specify data in multiple formats.
```

Comment:
Please add a table specifying the mapping between the ImageDataEncoding enum values and the associated typed array types.

Reply:
Updated.